From 9f6142b43b730d2ff51cf2b355050548f5f8fc9e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Tue, 13 Jan 2026 09:53:07 +0100
Subject: [PATCH 4/6] fakelib: add test for the new fileattr plugin

This is a package that "builds" by copying two files: a systemd
library that is known to use dlopen notes and another glibc library
that doesn't have them. It can be built with 'rpmbuild' or 'fedpkg local'.
For example:
  (cd fakelib && fedpkg local && echo 'Requires:' && rpm -qpv --requires x86_64/fakelib-0-1.fc44.x86_64.rpm && echo 'Recommends:' && rpm -qpv --recommends x86_64/fakelib-0-1.fc44.x86_64.rpm && echo 'Suggests:' && rpm -qpv --suggests x86_64/fakelib-0-1.fc44.x86_64.rpm)
---
 fakelib/fakelib.spec | 36 ++++++++++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)
 create mode 100644 fakelib/fakelib.spec

diff --git a/fakelib/fakelib.spec b/fakelib/fakelib.spec
new file mode 100644
index 0000000000..40714cbeb1
--- /dev/null
+++ b/fakelib/fakelib.spec
@@ -0,0 +1,36 @@
+# SPDX-License-Identifier: MIT-0
+#
+# This file is part of the package-notes package.
+
+Name:           fakelib
+Version:        0
+Release:        %autorelease
+Summary:        %{name}
+
+License:        None
+
+%define __dlopen_notes_requires_opts   --rpm-features=fakelib:gcrypt,fakelib:lz4
+%define __dlopen_notes_recommends_opts --rpm-features=*:zstd
+%define __dlopen_notes_suggests_opts   --rpm-features=fakelib:lzm[abc]
+
+#undefine _dlopen_notes_generator
+
+%description
+%{summary}.
+
+%prep
+
+%build
+
+%install
+install -Dt %{buildroot}/usr/lib64/ /usr/lib64/libsystemd.so.0
+ln -s libsystemd.so.0 %{buildroot}/usr/lib64/libsystemd.so
+install -Dt %{buildroot}/usr/lib64/ /usr/lib64/libmvec.so.1
+
+%files
+/usr/lib64/libsystemd.so.0
+/usr/lib64/libsystemd.so
+/usr/lib64/libmvec.so.1
+
+%changelog
+%autochangelog
